### ISSUE TITLE
[example] Implement Speech Recognition example.

### ISSUE THREAD

**Issue by BruceDai** [intel/webml-polyfill#924]

**Comment by miaobin** [comment]
We are now trying to add Speech Recognition based on  [Baidu's Deep Speech research paper](https://arxiv.org/abs/1412.5567).
When we convert the model to tflite. But we found that there are several ops that we have not supported yet: AudioSpectrogram, Mfcc, Minimum, StridedSlice, Split, Logistic.
![speech](https://user-images.githubusercontent.com/17840011/62597965-3f12dd80-b91a-11e9-8b7b-3d60e630a98a.png)

**Comment by miaobin** [comment]
I implemented the [Simple Audio Recognition](https://www.tensorflow.org/tutorials/sequences/audio_recognition) with webml-polyfill based on [KWS CNN](https://www.isca-speech.org/archive/interspeech_2015/papers/i15_1478.pdf). And I want to add a LSTM-esque model for this example. I have found a pb model from [KWS LSTM](https://github.com/ARM-software/ML-KWS-for-MCU/tree/master/Pretrained_models/LSTM). But when I convert the pb model to tflite I get the following err log:
```
Some of the operators in the model are not supported by the standard TensorFlow Lite runtime. 
If those are native TensorFlow operators, you might be able to use the extended runtime by 
passing --enable_select_tf_ops, or by setting target_ops=TFLITE_BUILTINS,SELECT_TF_OPS 
when calling tf.lite.TFLiteConverter(). Otherwise, if you have a custom implementation for them 
you can disable this error with --allow_custom_ops, or by setting allow_custom_ops=True when 
calling tf.lite.TFLiteConverter(). Here is a list of builtin operators you are using: ADD, 
CONCATENATION, FULLY_CONNECTED, LESS, LOGISTIC, MUL, RESHAPE, SPLIT, TANH, TRANSPOSE. 
Here is a list of operators for which you will need custom implementations: 
AudioSpectrogram, DecodeWav, Enter, Exit, LoopCond, Merge, Mfcc, Switch, TensorArrayReadV3, 
TensorArrayScatterV3, TensorArrayV3.
```

When I added the --allow_custom_ops flag to the same command above, I received the following output:
```
Exception: TensorFlow Lite currently doesn't support control flow ops: Enter, Exit, Merge, Switch.
```
It seems like that control flow is one of the biggest missing features of tflite. https://github.com/tensorflow/community/pull/83

**Comment by ibelem** [comment]
@miaobin Please list more support comparsion list for different platforms.

**Comment by ibelem** [comment]
@Also add other models need to be investigated.

**Comment by miaobin** [comment]
I can't find any information about AudioSpectrogram and MFCC from the website of the following platforms: [MKLDNN](http://intel.github.io/mkl-dnn/modules.html), [clDNN](https://intel.github.io/clDNN/activation_8hpp_source.html), [DirectML](https://docs.microsoft.com/en-us/windows/win32/direct3d12/dml-intro), [MPS](https://developer.apple.com/documentation/metalperformanceshaders) and [BNNS](https://developer.apple.com/documentation/accelerate/bnns). 
I can only find the information about [MFCC](https://www.tensorflow.org/api_docs/python/tf/signal/mfccs_from_log_mel_spectrograms?hl=en) from tensorflow.org and [tflite kernel](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/). Maybe we can convert them into WASM. BTW, I think we can use [stft](https://github.com/tensorflow/tfjs-core/pull/1746) and [abs](https://js.tensorflow.org/api/latest/#abs) to implement AudioSpectrogram with tfjs.

**Comment by miaobin** [comment]
I also tried an OpenVINO example from [Automatic Speech Recognition Sample](https://docs.openvinotoolkit.org/2018_R5/_samples_speech_sample_README.html). It is a model inference based on [Kaldi](https://kaldi-asr.org/) neural networks and speech feature vectors. I converted the model to Intel IR forma and ran the sample. The output is a Kaldi ARK file which stored the acoustic log likelihood sequences for all utterances.
